woocommerce/woocommerce-android
# Delete product categories

Part of: #11061

⚠️ Do not merge. This PR relies on https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2974 that need to be merged before.
### Description

Adds a new option to delete a product category. 
Additionally this PR also improves the progress dialog displayed while updating or deleting the product category. The title now matches the action the user is performing (add/update/remove). 
The snackbar for the success is also updated accordingly

### Testing instructions


1. Open the product detail screen
2. Select the category section
3. Make sure you have existing categories. If not, add a few of them.
4. Tap on any of the existing categories. Do not tap on the checkbox, tap on the list item
5. This will open the category "detail" 
6. Click on `Delete` button
7.  Notice the category is removed from the list successfully

### Images/gif


https://github.com/woocommerce/woocommerce-android/assets/2663464/c5531200-7a62-449a-a0f3-a68b4359b66b
